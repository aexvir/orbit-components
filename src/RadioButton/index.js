// @flow

import * as React from "react";
import { darken } from "polished";
import * as tokens from "orbit-design-token";

type Props = {
  name: string,
  label: string,
  value: string,
  error?: boolean,
  disabled: boolean,
  checked?: boolean,
  onChange: (SyntheticEvent<HTMLInputElement>) => any,
};

const RadioButton = (props: Props) => {
  const { name, label, value, error, disabled, checked, onChange } = props;

  let borderColor = tokens.borderColorInput;
  let iconColor = tokens.colorIconRadioButton;

  if (error && !disabled && !checked) {
    borderColor = tokens.borderColorInputError;
  } else if (disabled) {
    iconColor = tokens.colorIconRadioButtonDisabled;
  }
  return (
    <label htmlFor={value}>
      <input
        id={value}
        value={value}
        type="radio"
        name={name}
        disabled={disabled}
        checked={checked && "checked"}
        onChange={onChange}
      />
      <span className="radioBox">
        <span className="glyph" />
      </span>
      <span className="labelText">{label}</span>
      <style jsx>{`
        label {
          display: flex;
          flex-direction: row;
          align-items: center;
          height: ${tokens.heightCheckbox};
          opacity: ${disabled ? tokens.opacityCheckboxDisabled : "1"};
          cursor: ${disabled ? "default" : "pointer"};
          position: relative;
        }
        input[type="radio"] {
          visibility: hidden;
          display: none;
        }
        .radioBox {
          position: relative;
          box-sizing: border-box;
          display: flex;
          align-items: center;
          justify-content: center;
          background-color: ${tokens.backgroundInput};
          height: ${tokens.heightCheckbox};
          width: ${tokens.widthCheckbox};
          border-radius: ${tokens.borderRadiusCircle};
          overflow: hidden;
          border: 1px solid ${borderColor};
          transform: scale(1);
          transition: all ${tokens.durationFast} linear;
        }
        .radioBox .glyph {
          visibility: hidden;
          height: ${parseInt(tokens.heightCheckbox, 10) - 8}px;
          width: ${parseInt(tokens.heightCheckbox, 10) - 8}px;
          border-radius: ${tokens.borderRadiusCircle};
          background-color: ${iconColor};
        }
        input[type="radio"]:checked + .radioBox .glyph {
          visibility: visible;
        }
        label:focus {
          outline-width: 0;
        }
        label:focus .radioBox {
          outline-width: 0;
          box-shadow: 0 0 3px rgb(77, 144, 254);
        }
        label:hover .radioBox {
          border-color: ${disabled
            ? tokens.borderColorInput
            : darken(tokens.modifierDarkenHover, tokens.borderColorInput)};
        }
        .labelText {
          font-family: ${tokens.fontFamily};
          font-weight: ${tokens.fontWeightNormal};
          font-size: ${tokens.fontSizeLabelForm};
          color: ${tokens.colorLabelForm};
          margin-left: ${tokens.spaceXSmall};
        }
        input[type="radio"]:checked ~ .labelText {
          font-weight: ${tokens.fontWeightMedium};
        }
        label:active .radioBox {
          border-color: ${disabled
            ? borderColor
            : darken(tokens.modifierDarkenActive, tokens.borderColorInput)};
          ${!disabled ? `transform: scale(${tokens.modifierScaleButtonActive})` : ""};
        }
      `}</style>
    </label>
  );
};

RadioButton.defaultProps = {
  error: false,
  disabled: false,
  checked: false,
};

export default RadioButton;
