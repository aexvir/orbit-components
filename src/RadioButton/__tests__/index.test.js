// @flow

import * as React from "react";
import { shallow } from "enzyme";

import RadioButton from "../";

const label = "RadioButton";
const onChange = jest.fn();
const name = "radio";
const value = "button";

describe("Default RadioButton", () => {
  const component = shallow(
    <RadioButton label={label} onChange={onChange} name={name} value={value} />,
  );

  it("Should contain a label", () => {
    expect(component.find(".labelText").text()).toBe(label);
  });
  it("Input should not contain a disabled prop", () => {
    expect(component.find(`input[type="radio"]`).prop("disabled")).toBe(false);
  });
  it("Should not execute onChange method", () => {
    component.find("input").simulate("change");
    expect(onChange).toHaveBeenCalled();
  });
  it("Should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
