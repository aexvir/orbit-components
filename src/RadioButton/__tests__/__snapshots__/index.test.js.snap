// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default RadioButton Should match snapshot 1`] = `
<label
  className="jsx-502185451"
  htmlFor="button"
>
  <input
    checked={false}
    className="jsx-502185451"
    disabled={false}
    id="button"
    name="radio"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    type="radio"
    value="button"
  />
  <span
    className="jsx-502185451 radioBox"
  >
    <span
      className="jsx-502185451 glyph"
    />
  </span>
  <span
    className="jsx-502185451 labelText"
  >
    RadioButton
  </span>
  <JSXStyle
    css="label.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;height:20px;opacity:1;cursor:pointer;position:relative;}input[type=\\"radio\\"].__jsx-style-dynamic-selector{visibility:hidden;display:none;}.radioBox.__jsx-style-dynamic-selector{position:relative;box-sizing:border-box;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;background-color:#fff;height:20px;width:20px;border-radius:50%;overflow:hidden;border:1px solid #bac7d5;-webkit-transform:scale(1);-ms-transform:scale(1);transform:scale(1);-webkit-transition:all 0.2s linear;transition:all 0.2s linear;}.radioBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:hidden;height:12px;width:12px;border-radius:50%;background-color:#00a991;}input[type=\\"radio\\"].__jsx-style-dynamic-selector:checked+.radioBox.__jsx-style-dynamic-selector .glyph.__jsx-style-dynamic-selector{visibility:visible;}label.__jsx-style-dynamic-selector:focus{outline-width:0;}label.__jsx-style-dynamic-selector:focus .radioBox.__jsx-style-dynamic-selector{outline-width:0;box-shadow:0 0 3px rgb(77,144,254);}label.__jsx-style-dynamic-selector:hover .radioBox.__jsx-style-dynamic-selector{border-color:#aabacb;}.labelText.__jsx-style-dynamic-selector{font-family:\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif;font-weight:400;font-size:14px;color:#46515e;margin-left:8px;}input[type=\\"radio\\"].__jsx-style-dynamic-selector:checked~.labelText.__jsx-style-dynamic-selector{font-weight:500;}label.__jsx-style-dynamic-selector:active .radioBox.__jsx-style-dynamic-selector{border-color:#9aadc2;transform: scale(0.9);}"
    dynamic={
      Array [
        "20px",
        "1",
        "pointer",
        "#fff",
        "20px",
        "20px",
        "50%",
        "#bac7d5",
        "0.2s",
        12,
        12,
        "50%",
        "#00a991",
        "#aabacb",
        "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
        "400",
        "14px",
        "#46515e",
        "8px",
        "500",
        "#9aadc2",
        "transform: scale(0.9)",
      ]
    }
    styleId="4000472932"
  />
</label>
`;
